### ISSUE TITLE
Adapt CHECK_TILE_NEAR to check only absolute error

### ISSUE THREAD

**Issue by Sely85** [eth-cscs/DLA-Future#285]
Using relative error to check small numbers can be misleading, see https://github.com/eth-cscs/DLA-Future/pull/261#discussion_r515132060.  
While CHECK_MATRIX_NEAR can be used only to verify only absolute error, CHECK_TILE_NEAR macro does not work if only this error (not both) is inspected.  

_Originally posted by @albestro in https://github.com/eth-cscs/DLA-Future/pull/261#discussion_r514989781

**Comment by rasolca** [comment]
Only the documentation (preconditions) has to be updated, the macro already works:
- if you pass 0 to relative error, only the absolute error is checked,
- if you pass 0 to absolute error, only the relative error is checked.

**Comment by albestro** [comment]
Not really sure about that.

this is the tile utility

https://github.com/eth-cscs/DLA-Future/blob/52ea1a0f82e06193f74c28c1abc6e39f09a695ca/test/include/dlaf_test/matrix/util_tile.h#L164-L168

compared to matrix one

https://github.com/eth-cscs/DLA-Future/blob/52ea1a0f82e06193f74c28c1abc6e39f09a695ca/test/include/dlaf_test/matrix/util_matrix.h#L130-L135

**Comment by Sely85** [comment]
I gave it a try and it works. 
I updated the preconditions in `util_tile.h` to look like those in `util_matrix.h`:
```
void checkNear(ElementGetter expected, const Tile<T, Device::CPU>& tile, BaseType<T> rel_err,
               BaseType<T> abs_err, const char* file, const int line) {
  ASSERT_GE(rel_err, 0);
  ASSERT_GE(abs_err, 0);
  ASSERT_TRUE(rel_err > 0 || abs_err > 0);
```
Sometimes you may need to increase the "coefficient" of the absolute error. For example this one:
https://github.com/eth-cscs/DLA-Future/blob/52ea1a0f82e06193f74c28c1abc6e39f09a695ca/test/unit/test_blas_tile/test_gemm.h#L108-L109
needs to be re-written as: 
```
 CHECK_TILE_NEAR(res_c, c, 0,
                  20 * (k + 1) * TypeUtilities<T>::error);
```
to pass the test (coefficient increased from 2 to 20).

**Comment by rasolca** [comment]
I consider precondition asserts part of the doc.
checkNear can be fixed in a PR.
Anyway I don't see any reason to change the existing value tests to more coarser checks (As @Sely85 example).
Note that the two parameters check allows e.g.
relative error = 0.01 and absolute error 0.001, which means values with magnitude >= 0.1 are tested with the relative error while the others (abs(value) < 0.1) are tested with the absolute error.
This combines two advantages:
- floating point number have relative precision therefore larger number are less precise
- very small numbers can be the result of operations with larger numbers, and therefore the error might be larger.

**Comment by rasolca** [comment]
closed by #286
